recipes/lz4/all/conanfile.py

```diff
@@ -79,17 +80,48 @@ def build(self):
         cmake.configure(build_script_folder=self._cmakelists_folder)
         cmake.build()
 
+    def _create_cmake_module_alias_targets(self, module_file, targets):
+        content = ""
+        for alias, aliased in targets.items():
+            content += textwrap.dedent("""\
+                if(TARGET {aliased} AND NOT TARGET {alias})
+                    add_library({alias} INTERFACE IMPORTED)
+                    set_property(TARGET {alias} PROPERTY INTERFACE_LINK_LIBRARIES {aliased})
+                endif()
+            """.format(alias=alias, aliased=aliased))
+        save(self, module_file, content)
+
+    @property
+    def _module_file_rel_path(self):
+        return os.path.join("lib", "cmake", f"conan-official-{self.name}-targets.cmake")
+
     def package(self):
         copy(self, "LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
         cmake = CMake(self)
         cmake.install()
+        if Version(self.version) >= "1.9.4":
+            rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
         rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
         rmdir(self, os.path.join(self.package_folder, "share"))
 
+        if Version(self.version) >= "1.9.4":
+            # TODO: to remove in conan v2 once legacy generators removed
+            self._create_cmake_module_alias_targets(
+                os.path.join(self.package_folder, self._module_file_rel_path),
+                {"lz4": "LZ4::lz4" if self.options.shared else "LZ4::lz4_static"}
+            )
+
     def package_info(self):
         self.cpp_info.set_property("pkg_config_name", "liblz4")
+        if Version(self.version) >= "1.9.4":
+            self.cpp_info.set_property("cmake_file_name", "lz4")
+            self.cpp_info.set_property("cmake_target_name", "LZ4::lz4" if self.options.shared else "LZ4::lz4_static")
```

Comment:
not sure it's a good idea to have this condition.

Reply:
Also here I am lacking the necessary insight. Let me know if I should remove the condition or change it in any way.